.github/workflows/test.yaml

```diff
@@ -124,7 +126,7 @@ jobs:
       - name: Install tox
         run: python3 -m pip install tox
       - name: Run tests
-        run: tox --result-json=test-result.json
+        run: tox --result-json=test-result.json -c ${{ env.TOX_INI_DIR }}
```

Comment:
My initial thought after some tips from @jdkandersson was to use `working-directory` as either [default run arguments](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#defaultsrun) or for this specific run task.
However, this wasn't applicable to following tasks that `uses` rather than `runs`.
Since currently `tox.ini` file is the only file that requires being in another directory, I think it is simpler at the moment to direct tox to target the file rather than have the workflow run in another directory and change the tasks that run afterwards.
Please let me know if you have any other comments! 😃

Reply:
Or even just determine it more locally rather than defining an environment variable, we can't use a parameter since we want to rely on this workflow directly

Reply:
This could be the reason why a separate file for the `pull_request ` might be good so that it simulates exactly how it is used by external consumers

Reply:
I think having an optional parameter would be better for how we will be extending these tests in the future :)
Committed changes accordingly!